algo/src/main/kotlin/com/hartwig/actin/algo/evaluation/treatment/CurrentlyGetsChemoradiotherapyWithSpecificChemotherapyTypeAndMinimumCycles.kt

```diff
@@ -0,0 +1,65 @@
+package com.hartwig.actin.algo.evaluation.treatment
+
+import com.hartwig.actin.algo.evaluation.EvaluationFactory
+import com.hartwig.actin.algo.evaluation.EvaluationFunction
+import com.hartwig.actin.algo.evaluation.util.DateComparison
+import com.hartwig.actin.datamodel.PatientRecord
+import com.hartwig.actin.datamodel.algo.Evaluation
+import com.hartwig.actin.datamodel.clinical.treatment.TreatmentCategory
+import com.hartwig.actin.datamodel.clinical.treatment.TreatmentType
+import com.hartwig.actin.datamodel.clinical.treatment.history.TreatmentHistoryEntry
+import java.time.LocalDate
+
+class CurrentlyGetsChemoradiotherapyWithSpecificChemotherapyTypeAndMinimumCycles(
+    private val type: TreatmentType,
+    private val minCycles: Int,
+    private val referenceDate: LocalDate
+) : EvaluationFunction {
+
+    override fun evaluate(record: PatientRecord): Evaluation {
+        val treatmentMatches = record.oncologicalHistory.groupBy {
+            val matchingCategories = it.categories().containsAll(setOf(TreatmentCategory.CHEMOTHERAPY, TreatmentCategory.RADIOTHERAPY))
+
+            val latestStart = record.oncologicalHistory.maxOfOrNull { LocalDate.of(it.startYear ?: 0, it.startMonth ?: 1, 1) }
+            val enoughCyclesAndOngoingTreatment = enoughCyclesAndOngoingTreatment(it, latestStart)
+
+            when {
+                (matchingCategories &&
+                        it.isOfType(type) == true &&
+                        enoughCyclesAndOngoingTreatment == true) -> {
+                    true
+                }
+
+                (!matchingCategories &&
+                        it.categories().isNotEmpty() ||
+                        it.isOfType(type) == false ||
+                        enoughCyclesAndOngoingTreatment == false) -> {
+                    false
+                }
+
+                else -> {
+                    null
+                }
```

Comment:
The `when` conditions don't need parentheses, and we could drop the braces around the simple return values if they fit on the same line:
```suggestion
                matchingCategories && it.isOfType(type) == true && enoughCyclesAndOngoingTreatment == true -> true

                !matchingCategories && it.categories().isNotEmpty() || it.isOfType(type) == false || enoughCyclesAndOngoingTreatment == false -> false

                else -> null
```